students/eowyn/session03/mailroom.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+''' global variables'''
+donor_names = ["Margaret Atwood","Fred Armisen","Heinz the Baron Krauss von Espy"]
+donations= [[300,555],[240,422,1000],[1500,2300]]
+DONORS = list(zip(donor_names,donations))
+
+def mainloop():
+    ''' main interactive loop 
+    "send a thank you" "create a report" or "quit"
+    '''
+    while True:
+      answer = int(input("Select from one of these options: \n"
+          "(1) Send a Thank You\n"
+          "(2) Create a Report\n" 
+          "(3) Quit\n>"))
+      if answer == 3:
+        break
+      elif answer == 1:
+        thank_you()
+      elif answer == 2:
+        print_report()
+      else:
+        print("Please type 1, 2, or 3")
+
+def thank_you():
+    ''' Update donor records and print thank you notes.'''
+    while True:
+        fullname = input("To send a thank you, "
+            "select from one of these options: \n"
+            "Enter a donor name (new or existing)\n"
+            "Enter all to list existing DONORS \n"
+            "Enter menu to return to main menu \n >")
+        fullname = remove_inputquotes(fullname)
+        if fullname == 'menu':
+            break
+        if fullname.lower() == 'all':
+            print("All Donors:")
+            [print(x[0]) for x in DONORS]
+        else:
+            try: 
+                i = [x[0] for x in DONORS].index(fullname)
+                print("Existing Donor")
```

Comment:
Nice list comprehension. :) It is useful to only have one thing in a try block, that way you know precisely where the failure is. Note that you are writing a couple of lines of code twice, so no reason to have those in the try/except block anyway. You can use the else block to keep going. Example:
```
amount = float(input("Donation amount: "))
try:
    i = [x[0] for x in DONORS].index(fullname)
except ValueError:
   DONORS.append((fullname,[amount]))
   this_donor = DONORS[-1]
else:
    this_donor = DONORS[i]
    this_donor[1].append(amount)
print_letter(this_donor)    
```